src/Dependencies/Collections/SegmentedArray`1.cs

```diff
@@ -30,17 +31,38 @@ namespace Microsoft.CodeAnalysis.Collections
         /// including any padding the implementation chooses to add. Specifically, array elements lie <c>sizeof</c>
         /// bytes apart.</para>
         /// </remarks>
-        private static readonly int s_segmentSize = SegmentedArrayHelper.CalculateSegmentSize(Unsafe.SizeOf<T>());
+        private static int SegmentSize
+        {
+            [MethodImpl(SegmentedArrayHelper.FastPathMethodImplOptions)]
+            get
```

Comment:
>get [](start = 12, length = 3)

Use expression bodies?

Reply:
I tried using `[method: MethodImpl(...)]` with expression bodied properties, but it didn't compile.